satpy/tests/test_composites.py

```diff
@@ -173,83 +175,155 @@ def setUp(self):
         self.ds4 = ds4
 
         # high resolution version - but too big
-        ds4 = xr.DataArray(da.ones((4, 4), chunks=2, dtype=np.float64),
-                           attrs=attrs.copy(), dims=('y', 'x'),
-                           coords={'y': [0, 1, 2, 3], 'x': [0, 1, 2, 3]})
-        ds4.attrs['name'] += '4'
-        ds4.attrs['resolution'] = 500
-        ds4.attrs['rows_per_scan'] = 1
-        ds4.attrs['area'] = AreaDefinition('test', 'test', 'test',
-                                           {'proj': 'merc'}, 4, 4,
-                                           (-2000, -2000, 2000, 2000))
-        self.ds4_big = ds4
+        ds4_big = xr.DataArray(da.ones((4, 4), chunks=2, dtype=np.float64),
+                               attrs=attrs.copy(), dims=('y', 'x'),
+                               coords={'y': [0, 1, 2, 3], 'x': [0, 1, 2, 3]})
+        ds4_big.attrs['name'] += '4'
+        ds4_big.attrs['resolution'] = 500
+        ds4_big.attrs['rows_per_scan'] = 1
+        ds4_big.attrs['area'] = AreaDefinition('test', 'test', 'test',
+                                               {'proj': 'merc'}, 4, 4,
+                                               (-2000, -2000, 2000, 2000))
+        self.ds4_big = ds4_big
 
-    def test_bad_color(self):
+    @pytest.mark.parametrize(
+        "init_kwarg",
+        [
+            "bad",
+            "bad"
+        ]
+    )
+    def test_bad_colors(self, init_kwarg):
         """Test that only valid band colors can be provided."""
         from satpy.composites import RatioSharpenedRGB
-        self.assertRaises(ValueError, RatioSharpenedRGB, name='true_color', high_resolution_band='bad')
+        with pytest.raises(ValueError):
+            RatioSharpenedRGB(name='true_color', high_resolution_band="red", neutral_resolution_band=init_kwarg)
+            RatioSharpenedRGB(name='true_color', high_resolution_band=init_kwarg, neutral_resolution_band="red")
```

Comment:
I'm sorry, I don't understand what you're trying to do here. `"bad"` in the init_kwarg should be a dictionary of kwargs to pass to `RatioSharpenedRGB`. There should only be one call/creation of this class inside this single `raises`.

Reply:
my fault:sweat_smile: